src/services/services.ts

```diff
@@ -5617,6 +5620,10 @@ module ts {
         noRegexTable[SyntaxKind.TrueKeyword] = true;
         noRegexTable[SyntaxKind.FalseKeyword] = true;
 
+        // Just a stack of TemplateHeads and OpenCurlyBraces, used
+        // to perform rudimentary classification on templates.
+        var templateStack: SyntaxKind[] = [];

```

Comment:
Explain that rudimentary means 'inexact' and give example of where template classification will be wrong.